Change TeamFlag to fill a container instead of stretch

MiraiSubject opened the pull request:
Resolves #9963

Before: 

![image](https://user-images.githubusercontent.com/803255/91048727-5ce1dc00-e61c-11ea-84be-90dc32fed8d4.png)

![image](https://user-images.githubusercontent.com/803255/91048792-77b45080-e61c-11ea-9170-3c53f907994d.png)

After: 

![image](https://user-images.githubusercontent.com/803255/91048590-2c01a700-e61c-11ea-99e0-516384a275b2.png)

![image](https://user-images.githubusercontent.com/803255/91048902-a500fe80-e61c-11ea-9164-e569b8526a5b.png)

Visual tests using custom flags might be desired, but that would require custom assets (maybe in osu-resources?) to test the filling of the container. 

For now I've tested this locally by pointing `Texture` in `DrawableTeamFlag` to a flag in my `%localappdata%\osulazer\tournament\flags`

MiraiSubject committed:
Change TeamFlag from sprite to a container with a sprite

MiraiSubject committed:
Add scale to GroupTeam and remove unnecessary sizing and scaling in other scenes

MiraiSubject committed:
Merge branch 'master' into flag-fit

MiraiSubject committed:
Merge branch 'master' into flag-fit

MiraiSubject commented:
Updated the PR because GroupTeam missed its original size and in some scenes were some leftover size properties that could be replaced with scale or weren't needed at all any more.

I double checked all the scenes that use the flags using a custom flag and the built-in ones from the tests. For demonstration purposes, I've put screenshots of the different places that use them and also to show that the sizing, at least visually, hasn't changed. 

![dotnet_2020-09-13_20-01-57](https://user-images.githubusercontent.com/803255/93025160-07399780-f5fc-11ea-87fe-8489d871c239.png)

I made a lot of screenshots so I made it collapsible so it doesn't flood this comment section. 

<details>
<summary>Screenshots of various screens that use flags</summary>

TeamWinScreen

![image](https://user-images.githubusercontent.com/803255/93025193-3c45ea00-f5fc-11ea-9276-e99138aa8db4.png)

![image](https://user-images.githubusercontent.com/803255/93025234-8cbd4780-f5fc-11ea-9fb4-382a92e806e1.png)


TeamIntro

![image](https://user-images.githubusercontent.com/803255/93025204-4e278d00-f5fc-11ea-929c-9ae469e5b52c.png)

![image](https://user-images.githubusercontent.com/803255/93025238-92b32880-f5fc-11ea-96a3-2be7536585fc.png)


Gameplay

![image](https://user-images.githubusercontent.com/803255/93025212-5e3f6c80-f5fc-11ea-898d-3a19511f5f1e.png)

![image](https://user-images.githubusercontent.com/803255/93025242-9c3c9080-f5fc-11ea-818c-a93ef53e9667.png)

Schedule

![image](https://user-images.githubusercontent.com/803255/93025263-bc6c4f80-f5fc-11ea-903f-d3234e357dcd.png)

![image](https://user-images.githubusercontent.com/803255/93025253-a8285280-f5fc-11ea-84d7-6a2cf4227fc3.png)
</details>

peppy commented on osu.Game.Tournament/Screens/Ladder/Components/DrawableMatchTeam.cs:
```diff
@@ -63,7 +63,7 @@ public DrawableMatchTeam(TournamentTeam team, TournamentMatch match, bool losers
             this.losers = losers;
             Size = new Vector2(150, 40);
 
-            Flag.Scale = new Vector2(0.9f);
```
how did you calculate this one?

peppy commented on osu.Game.Tournament/Screens/Gameplay/Components/TeamDisplay.cs:
```diff
@@ -29,7 +29,6 @@ public TeamDisplay(TournamentTeam team, TeamColour colour, Bindable<int?> curren
             var anchor = flip ? Anchor.TopLeft : Anchor.TopRight;
 
             Flag.RelativeSizeAxes = Axes.None;
-            Flag.Size = new Vector2(60, 40);
```
won't this be bigger now?

peppy commented on osu.Game.Tournament/Screens/TeamWin/TeamWinScreen.cs:
```diff
@@ -90,8 +90,6 @@ private void matchChanged(ValueChangedEvent<TournamentMatch> match)
             {
                 new DrawableTeamFlag(match.Winner)
                 {
-                    Size = new Vector2(300, 200),
```
this is definitely wrong. check how it looks on master/release.

peppy commented:
If you need a reference, you can check any osulive stream vod (like [this one](https://www.twitch.tv/videos/739869464))

peppy commented:
@MiraiSubject planning to look into fixing this?

MiraiSubject commented:
> @MiraiSubject planning to look into fixing this?

Yes, I have some time again this week to look into it. Haven't been able to code much due to the start of university.

MiraiSubject committed:
Merge branch 'master' into flag-fit

MiraiSubject committed:
Changed scales of Seeding and Win screen to match the original

These were measured by pixel-to-pixel comparing master vs this branch in ShareX at the same resolution.

MiraiSubject committed:
Fixed gameplay flags being bigger and changed values to make more sense

MiraiSubject committed:
Merge branch 'master' into flag-fit

MiraiSubject committed:
Update flag scale of drawablematchteam

MiraiSubject commented:
I think scales and the values used in code make much more sense now. I used the size of `300, 200` and made it `75, 50` in `DrawableTeamFlag` and then calculated respective scales for the scenes that originally had sizes for the flags. 

One component I can't quite justify the `Scale` value for is `DrawableMatchTeam`

https://github.com/ppy/osu/blob/c24a29d1acdde0228ffdd24f7726e0a0100f9e16/osu.Game.Tournament/Screens/Ladder/Components/DrawableMatchTeam.cs#L66

I tried to get this one as accurate as possible by measuring the pixels at 1080p on `2020.1020.0` and this PR.

bdach commented:
It was closer, but not entirely without change yet. From what I can tell, for 100% matching the values should be:

| usage | `master` | current | should be to match |
| :- | :-: | :-: | :-: |
| `DrawableTeamFlag` | 70x47 | 75x50 | *could stay as baseline* |
| `DrawableTournamentTeam` (abstract) | 70x47 (default) | 75x50 (default) | 0.94 * 75x50 = 70.5x47 |
| `GroupTeam` | 36x24 | 75x50 * 0.5 = 37.5x25 | 75x50 * 0.48 = 36x24 |
| `TeamDisplay` | 60x40 | 75x50 * 0.8 = 60x40 | *correct* |
| `DrawableMatchTeam` | 45x30 * 0.9 = 40.5x27 | 75x50 * 0.55 = 41.25x27.5 | 75x50 * 0.54 = 40.5x27 |
| `SeedingScreen` | 300x200 * 0.3 = 90x60 | 75x50 * 1.2 = 90x60 | *correct* |
| `TeamWinScreen` | 300x200 * 0.5 = 150x100 | 75x50 * 2 = 150x100 | *correct* |

Overall I'm not a 100% sure this is the best method for specifying those sizes...

peppy committed:
Merge branch 'master' into flag-fit

peppy commented:
I think I'm going to go with 75x50 just because the ratio looks nice in code. Visually I've checked with OWC designs and the flags fit fine.

Will change 0.55 to 0.54 just because it's already a weird number (and 0.5 to 0.48 because the resultant sizing feels better).

peppy committed:
Adjust a couple of flag scales to match previous display size
